conans/model/build_info.py

```diff
@@ -109,8 +109,9 @@ def framework_paths(self):
             self._framework_paths = self._filter_paths(self.frameworkdirs)
         return self._framework_paths
 
-    def get_name(self, generator):
-        return self.names.get(generator, self.name)
+    def get_name(self, generator, pkg_name=None):
+        fallback_name = self.name if generator not in ["cmake", "cmake_multi"] else pkg_name  # FIXME: Remove in v1.22 (https://github.com/conan-io/conan/issues/6269#issuecomment-570182130)
```

Comment:
This should mimic the behavior that will be implemented by conan-center-index when recipes are fixed, so it should affect probably ``cmake_paths`` generator too.